### PR TITLE
ModelMaterialsCommon shader source is now in extras._pipeline

### PR DESCRIPTION
**Pull request by likangning93**
`modelMaterialsCommon` used to just put the shader source just in `extras.source`, which was preventing the rest of the pipeline from writing gltf that has new shaders (for example, models from the assimp converter) since the rest of the pipeline all works out of `extras._pipeline`.

Also updated a KHR_materials_common spec to check for this.

### CODE REVIEW

File: specs/lib/addDefaultsSpec.js [hunk 1]
```diff
@@ -244,6 +245,14 @@ describe('addDefaults', function() {
                 expect(gltfClone.materials[materialID].values).toEqual(expectValues);
             }
         }
+
+        // check that shader source is in extras._pipeline.source
+        var shaders = gltfClone.shaders;
+        expect(Object.keys(shaders).length > 0).toEqual(true);
+        for (var shaderID in shaders) {
+            var shader = shaders[shaderID];
```
Comment (lilleyse) [inline specs/lib/addDefaultsSpec.js]: When looping over keys, it's safer to check for `hasOwnProperty` like https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/master/lib/addDefaults.js#L19-L21.

File: specs/lib/addDefaultsSpec.js [hunk 2]
```diff
@@ -244,6 +245,14 @@ describe('addDefaults', function() {
...
+            expect(defined(shader.extras._pipeline.source)).toEqual(true);
```
Comment (lilleyse) [inline specs/lib/addDefaultsSpec.js]: Use `.toBeDefined()` instead.

**Comment by lilleyse** [comment]
This should be ready to go once https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/70 is in.

**Comment by likangning93** [comment]
Okay. Updated!

**Comment by lasalvavida** [comment]
@likangning93 You have jsHint failures

**Comment by likangning93** [comment]
I think those are outside my files though. Should I still address them here?

**Comment by lasalvavida** [comment]
Your branch has a committed half-merged file in `bin/gltf-pipeline.js` which is causing the failure.

**Comment by lasalvavida** [comment]
If you pull master and push I think it should fix it.

**Comment by likangning93** [comment]
Thanks Rob! Updated.

**Comment by lilleyse** [comment]
@likangning93 can you merge in master quickly?

**Comment by lasalvavida** [comment]
Looks good! Thanks @likangning93

**Pull request merged**